cmake/StaticBuild.cmake

```diff
@@ -23,7 +23,31 @@ if(BUILD_STATIC)
 
   # gettext is included in libc on many unix systems
   if(NOT LIBC_HAS_DGETTEXT)
-    set(GETTEXT_LIBRARIES "-Wl,-Bstatic -lintl -liconv -Wl,-Bdynamic")
+    FIND_LIBRARY(IDN2_LIBRARY NAMES idn2 libidn2
+      HINTS ${PC_GETTEXT_LIBDIR} ${PC_GETTEXT_LIBRARY_DIRS})
```

Comment:
I'm fairly sure idn2 was for GnuTLS, not gettext. What did your link errors look like?

Reply:
I'll look into this.  Would it be better for me to withdraw this pull request and submit more with smaller changes?

Reply:
You are correct.  I moved this portion of code.